LDMP/main_widget.py

```diff
@@ -54,10 +62,18 @@ def __init__(self, parent=None):
         super(MainWidget, self).__init__(parent)
 
         self.setupUi(self)
+
         # remove space before dataset item
         self.treeView_datasets.setIndentation(0)
         self.treeView_datasets.verticalScrollBar().setSingleStep(10)
 
+        # instantiate calcluate callback container to reuse most of old code
+        # do this before setting setupAlgorithmsTree
+        self.dlg_calculate_LD = DlgCalculateLD()
+        self.dlg_calculate_TC = DlgCalculateTC()
+        self.dlg_calculate_Biomass = DlgCalculateRestBiomass()
+        self.dlg_calculate_Urban = DlgCalculateUrban()
+
```

Comment:
This is very slow and introduces a noticeable delay in the UI from the moment the user presses the button to open the plugin dock until it is shown on screen. Please use a different strategy for loading these widgets without such a noticeable block of the main QGIS UI

Reply:
Yeah... I had to instantiate it before to use the click event wrapper of the interface. btw I can use a lambda so layer instantiate it.
Please open an issue about this

Reply:
I did lazy instantiation of the widget in the delegate (that is in charge of the run the callback) but I've some strange side effects... I'll give a look later. for now please leave it as it is and move the lazy instantiation in a separate issue.